src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -215,6 +215,13 @@ trait Implicits {
       tpeCache
     }
 
+    def dependsOnPrefix: Boolean = pre match {
+      case SingleType(pre0, _) => tpe.exists(_ =:= pre0)
+      case _ => false
+    }
```

Comment:
We'd need to think carefully about the performance here. Usually we need to cache the results of this sort of question.

Reply:
This would be called on a very infrequent execution path so I don't think it would have a measurable performance impact ... would a benchmark run improve your confidence level here?